src/tests/ftest/container/per_server_fault_domain.py

```diff
@@ -0,0 +1,301 @@
+        3. Stop the ranks_to_stop ranks.
+        4. Wait for rebuild to finish.
+        5. Get container's health status.
+        6. Restart the stopped servers for cleanup.
+        7. Verify container's Health status is expected.
+
+        Args:
+            create_pool (bool): Whether to create a pool.
+            ranks_to_stop (str): Comma-separated ranks to stop.
+            properties (str): Container property defined in the test yaml (rf_1 or rf_2).
+            expected_status: Expected container status at the end. HEALTHY or UNCLEAN.
+        """
+        # 1. Create a pool if create_pool is True. Then create a container with given
+        # container property.
+        if create_pool:
+            self.pool = self.get_pool()
+        self.container = self.get_container(pool=self.pool, properties=properties)
+
+        # Run IOR to write some data to the container.
+        self.ior_cmd.set_daos_params(
+            group=self.server_group, pool=self.pool, cont_uuid=self.container.uuid)
+        manager = self.get_ior_job_manager_command()
+        self.run_ior(manager=manager, processes=1)
+
+        # 4. Stop the ranks.
+        dmg_command = self.get_dmg_command()
+        dmg_command.system_stop(ranks=ranks_to_stop)
+
+        # 5. Wait for rebuild to finish.
+        self.pool.wait_for_rebuild_to_start(interval=10)
+        self.pool.wait_for_rebuild_to_end(interval=10)
+
+        # 6. Get container's health status.
+        get_prop_out = self.container.get_prop(properties=["status"])
+        status = get_prop_out["response"][0]["value"]
+
+        # 7. Restart the stopped servers for cleanup.
+        dmg_command.system_start(ranks=ranks_to_stop)
+
+        # 8. Verify container status.
+        self.assertEqual(
+            status, expected_status,
+            "Container status isn't {} with {}!".format(expected_status, properties))
+
+    def test_rf1_healthy(self):
+        """Jira ID: DAOS-11200
+
+        Test Description:
+        1. Determine the ranks to stop; two ranks in the same node.
+        2. Create a pool and a container with rf_lvl:2,rf:1
+        3. Run IOR to write some data to the container.
+        4. Stop the ranks.
+        5. Wait for rebuild to finish.
+        6. Get container's health status.
+        7. Restart the stopped servers for cleanup.
+        8. Verify container's Health status is HEALTHY.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=hw,medium
+        :avocado: tags=container
+        :avocado: tags=per_server_fault_domain,rf1_healthy
+        """
+        # 1. Determine the ranks to stop; two ranks in the same node.
+        # Select one host. (It doesn't matter which host to select.)
+        selected_host = list(set(self.server_managers[0].ranks.values()))[0]
+        ranks_to_stop = ",".join(
+            [str(rank) for rank, host in self.server_managers[0].ranks.items()
+             if host == selected_host])
```

Comment:
If we convert the dictionary of rank keys with host values to a dictionary of host keys with lists of ranks values then you can just pick a random host key to determine which list of ranks to kill:
```suggestion
        ranks_per_host = defaultdict(list)
        for rank, host in self.server_managers[0].ranks.items():
            ranks_per_host[str(host)].append(str(rank))
        host_to_stop = random.choice(list(dict(ranks_per_host).keys()))
        ranks_to_stop= ",".join(ranks_per_host[host_to_stop])
```

Reply:
Okay, thanks for the advice. However, the random host selection doesn't add any value to the rf_lvl:2 testing, so I believe selecting the first host is okay. (although the random selection would help making the test infrastructure more robust.)

Reply:
The current code does not select the first host.